### PR TITLE
Editing for the "What's new in TypeScript" page.

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Use the rich diff to compare

### CODE REVIEW

File: What's-new-in-TypeScript.md [hunk 1]
```diff
@@ -78,7 +84,7 @@ function f(x) { // Error: Not all code paths return a value.
 }
 ```
 
-### Case labels fall-through 
+### Case clause fall-throughs
 
 Reports errors for fall-through cases in switch statement. Check is turned *off* by default, and can enabled using `--noFallthroughCasesInSwitch`.
```
Comment (sandersn) [inline What's-new-in-TypeScript.md]: can **be** enabled

Comment (sandersn) [inline What's-new-in-TypeScript.md]: Is there a way to temporarily disable the check, like `// FALLTHROUGH` or something?

File: What's-new-in-TypeScript.md [hunk 2]
```diff
@@ -19,19 +19,22 @@ assign(x, { b: 10, d: 20 });
 assign(x, { e: 0 });  // Error
 ```
 
-## Unreachable code detection
+## Unreachable and unused code detection
```
Comment (mhegazy) [inline What's-new-in-TypeScript.md]: i would call it Control flow analysis or control flow errors

**Comment by mhegazy** [comment]
:+1:

File: What's-new-in-TypeScript.md [hunk 3]
```diff
@@ -105,63 +107,75 @@ Here are more examples of reachability checks in action:
...
-Module augmentation is a declaration of ambient module that directly nested either in external module or in top level ambient external module. 
+Users can now declare any augmentations that they want to make, or that any other consumers already have made, to an existing module.
+Module augmentations look like plain old ambient module declarations (i.e. the `declare module "foo" { }` syntax), and are directly nested either your own modules, or in another top level ambient external module.
 
-Name of module augmentation is resolved using the same set of rules as module specifiers in `import` \ `export` declarations. The declarations in module augmentation are merged with declarations inside module using standard rules for declaration merging. 
+Furthermore, TypeScript also has the notion of *global* augmentations of the form `declare global { }`.
+This allows modules to augment global types such as `Array` if necessary.
 
-Module augmentations cannot add new items to the top level scope but rather patch existing declarations.
+The name of module augmentation is resolved using the same set of rules as module specifiers in `import` and `export` declarations.
```
Comment (sandersn) [inline What's-new-in-TypeScript.md]: The name **of a** module augmentation

File: What's-new-in-TypeScript.md [hunk 4]
```diff
@@ -105,63 +107,75 @@ Here are more examples of reachability checks in action:
...
+The declarations in module augmentation are merged with any existing declarations the same way they would if they were declared in the same file.
```
Comment (sandersn) [inline What's-new-in-TypeScript.md]: in **a** module augmentation

File: What's-new-in-TypeScript.md [hunk 5]
```diff
@@ -369,13 +392,13 @@ else if (fso.isNetworked()) {
...
 
-Starting with TypeScript 1.8, official NuGet packages are available for the Typescript Compiler (tsc.exe) as well as the MSBuild integration (Microsoft.TypeScript.targets and Microsoft.TypeScript.Tasks.dll).
+Starting with TypeScript 1.8, official NuGet packages are available for the Typescript Compiler (`tsc.exe`) as well as the MSBuild integration (`Microsoft.TypeScript.targets` and `Microsoft.TypeScript.Tasks.dll`).
 
 Stable packages are available here:
 * [Microsoft.TypeScript.Compiler](https://www.nuget.org/packages/Microsoft.TypeScript.Compiler/)
 * [Microsoft.TypeScript.MSBuild](https://www.nuget.org/packages/Microsoft.TypeScript.MSBuild/)
 
-Also, a nightly NuGet packages to match the [nightly npm package](http://blogs.msdn.com/b/typescript/archive/2015/07/27/introducing-typescript-nightlies.aspx) is available on www.myget.org:
+Also, a nightly NuGet packages to match the [nightly npm package](http://blogs.msdn.com/b/typescript/archive/2015/07/27/introducing-typescript-nightlies.aspx) is available on https://myget.org:
```
Comment (sandersn) [inline What's-new-in-TypeScript.md]: a nightly NuGet **package** to match

File: What's-new-in-TypeScript.md [hunk 6]
```diff
@@ -1252,7 +1278,7 @@ for (var _i = 0, _a = expr; _i < _a.length; _i++) {
 ```
 
 ## Decorators
-> TypeScript decorator is based on the [ES7 decorator proposal](https://github.com/wycats/javascript-decorators). 
+> TypeScript decorator is based on the [ES7 decorator proposal](https://github.com/wycats/javascript-decorators).
```
Comment (sandersn) [inline What's-new-in-TypeScript.md]: TypeScript **decorators are** based on

**Comment by sandersn** [comment]
:+!:

File: What's-new-in-TypeScript.md (hunk 1 above)
Comment (DanielRosenwasser) [inline What's-new-in-TypeScript.md]: Not that I know of; the only exception is when the case is empty, which I'm documenting now.

**Comment by DanielRosenwasser** [comment]
Thanks guys!

**Pull request merged**